### PR TITLE
Add rgba2gray

### PR DESCRIPTION
**Pull request by rfezzani**
## Description

The `rgba2gray` function is added. It helps to fix/deprecate the RGBA image handling in `rgb2gray`.

## Checklist

- [x] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [x] Unit tests
- [x] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

File: skimage/color/colorconv.py [hunk 1]
```diff
@@ -745,6 +745,38 @@ def rgbcie2rgb(rgbcie):
     return _convert(rgb_from_rgbcie, rgbcie)
 
 
+def rgba2gray(rgba, background=(1, 1, 1)):
```
Comment (soupault) [inline skimage/color/colorconv.py]:
```suggestion
def rgba2gray(rgba, *, background=(1, 1, 1)):
```

**Comment by soupault** [comment]
I think that rgba2gray worth adding to the library, but, perhaps, there is a different opinion among the core devs.

**Comment by rfezzani** [comment]
`rgba2gray` alleviate the strange behavior of `rgb2gray` that accepts RGBA images and simply ignores the alpha channels:
https://github.com/scikit-image/scikit-image/blob/master/skimage/color/colorconv.py#L795

**Comment by jni** [comment]
@rfezzani but could we not fix `rgb2gray` with some keyword arguments instead? To me that is a nicer solution...

**Comment by rfezzani** [comment]
This PR combined with #4424 and #4418 is an attempt to solve #3185, ie add nD support to `colorconv` module. In fact, by defining `rgba2gray` and `gray2rgba`, we alleviate all the necessary guessing in `rgb2gray` and `gray2rgb` and it becomes easy to add nD support using `Ellipsis`...

**Comment by rfezzani** [comment]
@jni, to complete my previous answer, do your proposition looks like:

- the argument `background` set by default to `None` is added.
- if `background is not None and image.shape[:-1] == 4`, consider the input image as RGBA otherwise it is RGB.
- raise an error if `background is not None and image.shape[:-1] != 4`.

From the KISS principle point of view (one function -> one behavior) I am :-1:...

**Comment by jni** [comment]
@rfezzani simplicity scores very highly with me, but increasing the API surface area also increases complexity, albeit along a different axis. And in this case, because of the similarity of the function names, it makes the API harder to use from the standpoint of autocompletion.

I'm also wary of adding new color manipulation functions right now, before dealing with #4294. How do you feel about putting this on hold and revisiting it once we have changed that?

**Comment by rfezzani** [comment]
In my mind, RGB != RGBA, that's why it sounds not natural for me to have a function called `rgb2something` accepting RGBA images.

If the solution of adding a keyword to deal with alpha channel in `rgb2gray` is acceptable for you, there is no reason for not implementing it for all the `rgb2something` functions.

An other option would be to not implement the `rgba2gray` function (close this PR), but deprecate the RGBA support in `rgb2gray` and advise to preprocess RGBA images with `rgba2rgb` before any manipulation.

**Comment by jni** [comment]
> An other option would be to not implement the rgba2gray function (close this PR), but deprecate the RGBA support in rgb2gray and advise to preprocess RGBA images with rgba2rgb before any manipulation.

I don't mind this idea. Let's wait to see what others think!

**Comment by rfezzani** [comment]
> And in this case, because of the similarity of the function names, it makes the API harder to use from the standpoint of autocompletion.

I'm not sure it could be worse: Capture du 2020-01-31 10-27-13
:rofl:

**Comment by alexdesiqueira** [comment]
Hey everyone,
I'm kindly against implementing somehting like that. Since this function is "only" calling `rgb2gray(rgba2rgb(rgba, background))`, I think a note on `rgb2gray` and an illustrated example should suffice. What do you think?

**Comment by rfezzani** [comment]
The main goal of this PR is the deprecation of the RGBA support from `rgb2gray` by proposing an alternative! This deprecation considerably ease adding the nD support for the `colorconv` module like  in #4418.
@alexdesiqueira's solution is perfectly fine for me.

**Comment by rfezzani** [comment]
Please see  this [solution](https://github.com/scikit-image/scikit-image/pull/4418/files#diff-9f4dedd98e7cca606dec9f0b792c1d94R820) without `rgba2gray` function definition.

**Comment by rfezzani** [comment]
May I close this PR?

**Comment by rfezzani** [comment]
Closing :-)

**Pull request closed without merging**